docs/current_docs/api/clients-sdk.mdx

```diff
@@ -0,0 +1,418 @@
+- `go.mod`/`go.sum` manage the Go module and its dependencies.
+- `main.go` is where your Dagger module code goes. It contains sample code to help you get started.
+- `internal` contains automatically-generated types and helpers needed to configure and run the module:
+    - `dagger` contains definitions for the Dagger API that's tied to the currently running Dagger Engine container.
+    - `querybuilder` has utilities for building GraphQL queries (used internally by the `dagger` package).
+    - `telemetry` has utilities for sending Dagger Engine telemetry.
+
+</TabItem>
+<TabItem value="Python">
+
+```
+.
+├── LICENSE
+├── dagger
+│   ├── pyproject.toml
+│   ├── requirements.lock
+│   ├── sdk
+│   └── src
+│       └── main
+│           └── __init__.py
+└── dagger.json
+```
+
+In this structure:
+
+- `dagger.json` is the Dagger module configuration file.
+- `pyproject.toml` manages the Python project configuration.
+- `requirements.lock` manages the module dependencies.
+- `src/main/` is where your Dagger module code goes. It contains sample code to help you get started.
+- `sdk/` contains the Python SDK.
+
+This structure hosts a `main` package inside a `src` directory. This follows a [Python convention](https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/) that requires a project to be installed in order to run its code. This prevents accidental usage of development code since the Python interpreter includes the current working directory as the first item on the import path (more information is available in this [blog post on Python packaging](https://blog.ionelmc.ro/2014/05/25/python-packaging/)).
+
+</TabItem>
+<TabItem value="TypeScript">
+
+```
+.
+├── LICENSE
+├── dagger
+│   ├── package.json
+│   ├── sdk
+│   ├── src
+│   │   └── index.ts
+│   └── tsconfig.json
+└── dagger.json
+```
+
+In this structure:
+
+- `dagger.json` is the Dagger module configuration file.
+- `package.json` manages the module dependencies.
+- `src/` is where your Dagger module code goes. It contains sample code to help you get started.
+- `sdk/` contains the TypeScript SDK.
+
+</TabItem>
+</Tabs>
+
+:::note
+While you can use the utilities defined in the automatically-generated code above, you *cannot* edit these files. Even if you edit them locally, any changes will not be persisted when you run the module.
+:::
+
+You can now write Dagger Functions using the selected Dagger SDK. Here is an example, which calls a remote API method and returns the result:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+```go file=./snippets/functions/functions-complex/go/main.go
+```
+
+This Dagger Function includes the context as input and error as return in its signature.
+</TabItem>
+<TabItem value="Python">
+```python file=./snippets/functions/functions-complex/python/main.py
+```
+
+Dagger Functions are implemented as [@dagger.function][@function] decorated
+methods, of a [@dagger.object_type][@object_type] decorated class.
```

Comment:
The URLs for these reference-style links have been lost. When you see a `[name][ref]`, there should be a corresponding `[ref]: url` in its own line.

See https://daringfireball.net/projects/markdown/syntax#link

Reply:
Thanks for catching this, added them back